File: Python3/201 - 400/374 Guess Number Higher or Lower.py
# 30ms, 85.21%
class Solution(object):
    def guessNumber(self, n):
        beg = 1
        while beg <= n:
            mid = (beg + n) // 2
            res = guess(mid)
            if res > 0:
                beg = mid + 1
            elif res < 0:
                n = mid - 1
            else:
                return mid
        

File: 201 - 400/374 Guess Number Higher or Lower.cc
// 3ms, 21.34%
int guess(int num);

class Solution {
public:
    int guessNumber(int n) {
        int beg = 1;
        while (beg <= n) {
            int mid = beg + (n - beg) / 2;
            int res = guess(mid);
            if (res > 0)
                beg = mid + 1;
            else if (res < 0)
                n = mid - 1;
            else
                return mid;
        }
    }
};
